### PR TITLE
WAT2-24 - features/wat2-24-combine-events-into-single-contract

### PR DESCRIPTION
**Pull request by smallbatch-apps**

### CODE REVIEW

File: contracts/Licences.sol [hunk 1]
```diff
@@ -30,15 +31,18 @@ contract Licences is Ownable {
     mapping(bytes32 => bytes32) public _waterAccountIdToIdentifier;
     mapping(address => mapping(bytes32 => bytes32)) public _addressToZoneToWaterAccountId;
 
-    constructor() Ownable() {
+    OrderBook private _orderbook;
```
Comment (bitcoinbrisbane) [inline contracts/Licences.sol:34]:
```suggestion
    OrderBook immutable private _orderbook;
```

File: contracts/Zones.sol [hunk 2]
```diff
@@ -41,12 +31,22 @@ contract Zones is Ownable {
 
     function getZones() public view returns (Zone[] memory) {
         Zone[] memory zonesArray = new Zone[](zoneList.length);
-        for(uint8 i = 0; i < zoneList.length; i++) {
+        for (uint8 i = 0; i < zoneList.length; i++) {
```
Comment (bitcoinbrisbane) [inline contracts/Zones.sol:34]: To save gas so you dont need to SLOAD on each loop...
```suggestion
        uint256 count = zoneList.Length;
        for (uint8 i = 0; i < count; i++) {
```

**Review by bitcoinbrisbane — COMMENTED** [review]
Few suggestions to save gas

File: contracts/Licences.sol (hunk 1 above)
Comment (smallbatch-apps) [reply contracts/Licences.sol:34]: LIES!

`TypeError: Immutable variables cannot have a non-value type.`

Comment (smallbatch-apps) [reply contracts/Licences.sol:34]: Correction, you could put it there, but not in a few other places. I tried to use it too widely and made the compiler very unhappy.

**Review by aditi2313 — APPROVED** [review]
Looks good to me!

**Review by thanapat-civicledger — APPROVED** [review]
@smallbatch-apps 
The changes look good to me. However, will we create a new story to make all the disabled tests back to work again?

**Pull request merged**